.github/workflows/check.yml

```
@@ -32,10 +32,9 @@ jobs:
       - uses: actions/cache@v3
         with:
           path: |
-            patches
             node_modules
             */*/node_modules
-          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock', '**/patches') }}
```

i think what we actually want is to fetch fresh node modules if the yarn.lock OR any patches change. without taking into account the patches, PRs like [this](https://github.com/valora-inc/wallet/pull/3656) would fail - examples of failures are in the CI in https://github.com/valora-inc/wallet/pull/3651.

it seems like `path` was the wrong place to put the patches though. according to the [docs](https://github.com/actions/cache#inputs) this would mean that we are caching the patches, which is unnecessary. it also causes the wrong patch to be applied (e.g. https://github.com/valora-inc/wallet/actions/runs/4779094863/jobs/8495947030)

it seems [possible](https://docs.github.com/en/actions/learn-github-actions/expressions#example-with-multiple-patterns) to add multiple things to the `hashFiles` function. i tested this change by seeing that the [CI still passes](https://github.com/valora-inc/wallet/actions/runs/4779175382/attempts/1) -> [rerunning the same job and verifying that the cache is applied](https://github.com/valora-inc/wallet/actions/runs/4779175382/jobs/8496118656) -> creating a commit that just changes a patch file and verifying that [the cache is not hit](https://github.com/valora-inc/wallet/actions/runs/4779203159/jobs/8496132136?pr=3667), fresh node modules are fetched and the job still runs.

i left this change here because it is small and directly affecting this PR, if someone feels strongly then i can move it out to a separate PR. FYI @MuckT

---

Also proposing to match only the patches at the root of the repo ([docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-file-paths)):
```suggestion
          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock', 'patches/*.patch') }}
```